src/pages/execution/workflowTabs/gantt-chart/ConductorTimeline.tsx

```diff
@@ -0,0 +1,274 @@
+            if (subTaskIndex === -1){
+              subTaskData.push(initialData[i]);
+            }else{
+              subTaskData[subTaskIndex] = {...subTaskData[subTaskIndex], data: [...subTaskData[subTaskIndex].data, initialData[i].data[0]]};
+            }
+            i++;
+          }
+          data.push(...subTaskData);
+        }
+      }
+    })
+    return data;
+  }, [data]);
+  /** ID of Tasks which exist in fully collapsed view (may or may not have subtasks) */
+  const parentTaskIds = useMemo<string[]>(()=> initialData.filter((task) => {
+    const {parentTaskReferenceName:parentTaskRefName} = task
+    if (!parentTaskRefName || !collapseTaskTypes.includes(taskTypeMap.get(parentTaskRefName))){
+      return true;
+    }
+    return false;
+  }).map(task=>task.id)
+  , [collapsedData]);
+  /** Map of Task IDs to their content when collapsed */
+  const collapsedTaskMap = useMemo<Map<string, Series[]>>(()=> {
+    let subTaskMap = new Map<string, Series[]>();
+    parentTaskIds.forEach(taskId => {
+      let taskIdx = idToIndexMap.get(taskId)
+      let task = initialData[taskIdx];
+      subTaskMap.set(taskId, [task]);
+      let i = taskIdx+1;
+      let subTaskArr = subTaskMap.get(taskId);
+      while (i<initialData.length && initialData[i].parentTaskReferenceName === task.referenceTaskName){        
+        if (task.taskType === DO_WHILE){
+          let idx = subTaskArr.findIndex((subTask)=>subTask.referenceTaskName === initialData[i].referenceTaskName);
+          if (idx === -1){
+            subTaskArr.push(initialData[i]);
+          }else{
+            subTaskArr[idx] = {...subTaskArr[idx], data: [...subTaskArr[idx].data, initialData[i].data[0]]};
+          }
+        }else if (task.taskType === FORK_JOIN_DYNAMIC || task.taskType === FORK){
+          let newTime:Date = task.data[0].t2;
+          let oldTime:Date = subTaskArr[0].data[0].t2;
+          if (oldTime < newTime){
+            subTaskArr[0] = {...subTaskArr[0], data: [{...subTaskArr[0].data[0], t2: newTime}]}
+          }
+        }
+        i++;
+      } 
+      subTaskMap.set(taskId, subTaskArr);
+    })
+    return subTaskMap;
+  }, [collapsedData])
+  /** Map of Task IDs to their content when expanded */
+  const expandedTaskMap = useMemo<Map<string, Series[]>>(()=> {
+    let subTaskMap = new Map<string, Series[]>();
+    parentTaskIds.forEach(taskId => {
+      let taskIdx = idToIndexMap.get(taskId)
+      let task = initialData[taskIdx];
+      subTaskMap.set(taskId, [task]);
+      let i = taskIdx+1;
+      let subTaskArr = subTaskMap.get(taskId);
+      while (i<initialData.length && initialData[i].parentTaskReferenceName === task.referenceTaskName){
+        subTaskArr.push(initialData[i]); 
+        i++;
+      }    
+      subTaskMap.set(taskId, subTaskArr);
+    })
+    return subTaskMap;
+  }, [data])
+
+  const seriesMax:() => Date = ()=>{
```

Comment:
use `function()` notation for consistency with the other embedded functions, also this allows simplifies typing.